docs: add comparisons to automagic iamlive and actionhero tools

agilgur5 opened the pull request:
## What does this PR do?

[//]: # (Required: Describe the effects of your pull request in detail. If
multiple changes are involved, a bulleted list is often useful.)

adds [Action Hero](https://github.com/princespaghetti/actionhero) and [`iamlive`](https://github.com/iann0036/iamlive) tools to a new section in the Comparison docs
 
actionhero:
-  actionhero is a tool that actually _can_ automagically sense your
    AWS API calls
    - it is a <100 LoC sidecar that uses the AWS SDK's Client Side
      Monitoring (CSM) feature to monitor the AWS API calls that a running
      application makes
    - it does have several limitations due to using CSM though, as listed
      in the comparisons doc

- Very simple, under-the-radar tool that can be a great addition


iamlive:
- iamlive is another tool that can automagically sense AWS API calls
   - iamlive can make use of CSM or run as an embedded proxy
   - it does have similar limitations to other tools in that it relies
     on an unofficial mapping, as mentioned in the comparison
        
## What gif best describes this PR or how it makes you feel?

[//]: # (Encouraged: Insert an appropriate gif/meme to brighten up your reviewer's day.)

![giphy](https://user-images.githubusercontent.com/4970083/154814878-f3384cc3-79a2-405f-ae8f-fb0d4682cfe1.gif)

[Source](https://giphy.com/gifs/movie-guy-breath-BeLYLrX37eE6c). Return to monke? 🙈  

## Completion checklist

- [x] Additions and changes have unit tests (N/A)
- [ ] [Unit tests, Pylint, security testing, and Integration tests are passing.](https://github.com/salesforce/policy_sentry/actions). GitHub actions does this automatically
- [ ] The pull request has been appropriately labeled using the provided PR labels

agilgur5 commented:
I signed the CLA but upon submission it said some error occurred and to send an email. I tried refreshing and resubmitting but now it just says I've already signed 😐

agilgur5 commented:
My other PR (#397) succeeded on the CLA check 🤷

kmcquade commented on README.md:
```diff
@@ -40,7 +40,7 @@ Writing security-conscious IAM Policies by hand can be very tedious and ineffici
  * Determined to make your best effort to give users and roles the least amount of privilege you need to perform your duties, you spend way too much time combing through the AWS IAM Documentation on [Actions, Resources, and Condition Keys for AWS Services][1].
  * Your team lead encourages you to build security into your IAM Policies for product quality, but eventually you get frustrated due to project deadlines.
- * You don't have an embedded security person on your team who can write those IAM policies for you, and there's no automated tool that will automagically sense the AWS API calls that you perform and then write them for you in a least-privilege manner.
+ * You don't have an embedded security person on your team who can write those IAM policies for you, and there's no automated tool that will [automagically sense the AWS API calls that you perform](https://github.com/princespaghetti/actionhero) and then write them for you in a least-privilege manner.
```
Nit: I don't want to include that as a link here as it detracts from the point.

I did look at ActionHero back then, but my issue with it was that (1) it missed a lot of API Calls (this is AWS's fault because there is not a 1-1 mapping between CloudTrail event names, API call names, IAM namespaces) and (2) it doesn't include resource ARNs.

Since then, Ian McCay's [iamlive](https://github.com/iann0036/iamlive) tool has come out which has a ridiculously accurate mapping of the IAM/CloudTrail/API names, and even identifies undocumented APIs. I **love** iamlive. However, it still doesn't support resource ARNs due to limitations of AWS Client side monitoring.

Therefore, I think it would be fair to suggest that we change this to "there's no automated tool that will automagically sense the AWS API calls that you perform and then write them for you with resource ARN constraints." I can approve that if you make that change :)

agilgur5 replied:
I actually just found `iamlive` last night after looking for more CSM tools so didn't know of it when I wrote the PR! Agreed with changing the wording given those details now and not having a link

agilgur5 replied:
modified wording as discussed

kmcquade commented on docs/appendices/comparison-to-related-tools.md:
```diff
@@ -138,6 +138,26 @@ there are some major limitations:
         not every CloudTrail action matches IAM Actions, the results are
         not always accurate.
 
+Monitoring AWS API Calls
```
This is great! Updated comparisons to existing tools is very overdue. Really appreciate it.

agilgur5 replied:
Added `iamlive` in the comparison docs as well, let me know what you think!

Also let me know if there's any tools you'd like to add here that you haven't had time to, I might have some cycles to spare on adding those to this doc

kmcquade commented on docs/appendices/comparison-to-related-tools.md:
```diff
@@ -138,6 +138,26 @@ there are some major limitations:
+Action Hero is a sidecar style utility to monitor the AWS API calls that your
+application makes and log out the corresponding IAM actions.
+
+Action Hero is limited in that it cannot capture parameters and does not
```
Nit: It doesn't capture parameters *or* resource ARNs (due to limitations from AWS's Client-Side Monitoring)

agilgur5 replied:
By "parameters" I think they meant ARNs too. But yea last night I thought it would be good to be more specific about that too

agilgur5 replied:
included resource ARNs in wording

kmcquade reviewed (changes requested):
Thanks for the contribution! I really appreciate it. 😀

Happy to approve after the comments are addressed.

agilgur5 commented:
Addressed all comments and also added `iamlive` to the Comparison docs in the same section. Let me know what you think!

(P.S. seems like the CLA check updated now with the new commits)

kmcquade commented on docs/introduction.md:
```diff
@@ -7,7 +7,7 @@ Writing security-conscious IAM Policies by hand can be very tedious and ineffici
  * Determined to make your best effort to give users and roles the least amount of privilege you need to perform your duties, you spend way too much time combing through the AWS IAM Documentation on [Actions, Resources, and Condition Keys for AWS Services][1].
  * Your team lead encourages you to build security into your IAM Policies for product quality, but eventually you get frustrated due to project deadlines.
- * You don't have an embedded security person on your team who can write those IAM policies for you, and there's no automated tool that will automagically sense the AWS API calls that you perform and then write them for you in a least-privilege manner.
+ * You don't have an embedded security person on your team who can write those IAM policies for you, and there's no automated tool that will [automagically sense the AWS API calls that you perform](https://github.com/princespaghetti/actionhero) and then write them for you in a least-privilege manner.
```
I'll modify this sentence to reflect the changes in the README after merge so it is consistent.

agilgur5 replied:
Ah my bad I forgot this part is duplicated in two files, sorry for the oversight.
See you fixed it in https://github.com/salesforce/policy_sentry/commit/cd49c1bd0fccc7eb9ea9961ae6dfe8186c2887a8 , thanks for taking care of that!

kmcquade commented:
Great! thanks so much @agilgur5
